index.d.ts

```diff
@@ -1213,6 +1213,26 @@ declare module 'mongoose' {
     lean?: boolean;
   }
 
+  interface GeoNear {
+      /** [`$geoNear` reference](https://docs.mongodb.com/manual/reference/operator/aggregation/geoNear/) */
+      $geoNear: {
+      near: { type: 'Point'; coordinates: [number, number] } | [number, number];
+      distanceField: string;
+      distanceMultiplier?: number;
+      includeLocs?: string;
+      key?: string;
+      maxDistance?: number;
+      minDistance?: number;
+      query?: AnyObject;
+      spherical?: boolean;
+      /**
+       * Deprecated. Use only with MondoDB below 4.2 (removed in 4.2)
+       * @deprecated
+       */
+      num?: number;
```

Comment:
Also please add back `uniqueDocs?: boolean`, because Mongoose 5 supports versions of MongoDB that still support that version